src/main/kotlin/no/nav/amt/deltaker/bff/deltaker/api/PameldingApi.kt

```diff
@@ -0,0 +1,132 @@
+package no.nav.amt.deltaker.bff.deltaker.api
+
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.call
+import io.ktor.server.auth.authenticate
+import io.ktor.server.request.header
+import io.ktor.server.request.receive
+import io.ktor.server.response.respond
+import io.ktor.server.routing.Routing
+import io.ktor.server.routing.delete
+import io.ktor.server.routing.post
+import no.nav.amt.deltaker.bff.application.plugins.getNavAnsattAzureId
+import no.nav.amt.deltaker.bff.application.plugins.getNavIdent
+import no.nav.amt.deltaker.bff.auth.TilgangskontrollService
+import no.nav.amt.deltaker.bff.deltaker.DeltakerService
+import no.nav.amt.deltaker.bff.deltaker.PameldingService
+import no.nav.amt.deltaker.bff.deltaker.api.model.PameldingRequest
+import no.nav.amt.deltaker.bff.deltaker.api.model.PameldingUtenGodkjenningRequest
+import no.nav.amt.deltaker.bff.deltaker.api.model.UtkastRequest
+import no.nav.amt.deltaker.bff.deltaker.api.model.toDeltakerResponse
+import no.nav.amt.deltaker.bff.deltaker.model.GodkjenningAvNav
+import no.nav.amt.deltaker.bff.deltaker.model.OppdatertDeltaker
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+fun Routing.registerPameldingApi(
+    tilgangskontrollService: TilgangskontrollService,
+    deltakerService: DeltakerService,
+    pameldingService: PameldingService,
+) {
+    val log = LoggerFactory.getLogger(javaClass)
+
+    authenticate("VEILEDER") {
+        post("/pamelding") {
+            val navIdent = getNavIdent()
+            val pameldingRequest = call.receive<PameldingRequest>()
+            val enhetsnummer = call.request.header("aktiv-enhet")
+            tilgangskontrollService.verifiserSkrivetilgang(getNavAnsattAzureId(), pameldingRequest.personident)
+            val deltaker = deltakerService.opprettDeltaker(
+                deltakerlisteId = pameldingRequest.deltakerlisteId,
+                personident = pameldingRequest.personident,
+                opprettetAv = navIdent,
+                opprettetAvEnhet = enhetsnummer,
+            )
+            call.respond(deltaker.toDeltakerResponse())
+        }
+
+        // Deprecated bør byttes til /pamelding
+        post("/deltaker") {
```

Comment:
Du kan bare fjerne den gamle, så får vi sørge for at frontend blir oppdatert :) Bør metoden den kaller også flyttes til PameldingService? 

Ulempen med å endre til /pamelding er forsåvidt at apiet er ment å brukes for enten å starte påmelding eller å hente en allerede påmeldt deltaker, så da kan det være misvisende at apiet heter /pamelding, men jeg tror det er vanskelig å finne et perfekt navn her.. Synes det du foreslår her ser bra ut :)

Reply:
Henter man en opprettet deltaker med et `PameldingRequest`? Jeg tenkte i allefall at dette alltid ville skje i konteksten av en påmelding, og ikke etter, men jeg husker nok ikke hva vi kom frem til.

Det er litt kronglete å bestemme hvor denne bør høre hjemme, men synes selve opprettelsen av deltakeren bør skje i `DeltakerService` så slipper man å dra inn repository og annet hit.

Reply:
Også tenker jeg at det er best å beholde det gamle endepunktet litt, så får man oppdatere frontend litt når det passer, så andre oppgaver ikke blir hengende på vent.

Reply:
PameldingRequest inneholder jo bare fnr og deltakerlisteId, så ja, den brukes for hent eller opprett 😅 Men siden requesten heter det så kan jo like godt apiet hete /pamelding også. 

Enig i at den kan være begge steder, men ser poenget med å ikke måtte ha repositoryet inn i PameldingService. Sikkert best sånn det er nå :)

Reply:
Jeg prøver måten du foreslår, det fjerner jo litt mer fra deltakerservicen, så den blir litt enklere også. Får se hvordan det blir 👍

Reply:
Synes det ble bedre slik du foreslo 👍  Ryddigere å ha alt med kladd å gjøre `PameldingService`.